lib/number.js

```diff
@@ -192,41 +180,23 @@ function formatDecimal(value, options) {
 	options = validateFormatOptions(options);
 
 	const isNegative = value < 0;
-
-	const precisionScaling = Math.pow(10, options.maximumFractionDigits);
-	// round to desired precision
-	value = Math.abs(Math.round(value * precisionScaling) / precisionScaling);
-
-	const integerValue = Math.floor(value);
-
-	let ret = formatPositiveInteger(integerValue, descriptor);
-
-	const hasDecimal = value !== integerValue;
-	let decimalStr = null;
-
-	if (hasDecimal) {
-		let decimalValue = (value * precisionScaling - integerValue * precisionScaling) / precisionScaling;
-		decimalValue =  Math.round(decimalValue * precisionScaling) / precisionScaling;
-
-		// get a string of 0.xxx, or exponent of x.xe-x
-		decimalStr = '' + decimalValue;
-
-		if (decimalValue.toExponential() === decimalStr) {
-			// Get a string with leading zeros
-			decimalStr = formatExponentialDecimal(decimalStr);
-		} else {
-			// the first decimal place is index 2
-			decimalStr = decimalStr.slice(2);
+	value = Math.abs(value);
+
+	const strValue = new Intl.NumberFormat(
+		'en-US',
+		{
+			signDisplay: 'never',
+			maximumFractionDigits: options.maximumFractionDigits,
+			minimumFractionDigits: options.minimumFractionDigits,
+			useGrouping: false
 		}
-	} else if (options.minimumFractionDigits > 0) {
-		decimalStr = '';
-	}
+	).format(value);
 
-	if (decimalStr !== null) {
-		while (decimalStr.length < options.minimumFractionDigits) {
-			decimalStr += '0';
-		}
-		ret += descriptor.symbols.decimal + decimalStr;
+	let ret = formatPositiveInteger(parseInt(strValue), descriptor);
```

Comment:
If we didn't need to support our crazy grouping formats (the groups can be different sizes) this could be simplified further.

Reply:
I guess the same goes for the negative eh?

Reply:
Yes, although swapping in the negative symbol is pretty easy... but yeah I'd love to just rely on the native API even for that. Maybe one day we can do an analysis of what the native API produces and see if we can leverage it... the trick is you need to pass it a single locale that it'll use for the group, decimal AND negative symbol, and we let users customize all 3 of those things separately. But maybe for each of those 9 combinations there's a single locale we could map them to, who knows!